src/NServiceBus.Gateway.AcceptanceTests/When_sending_fails_with_retries.cs

```diff
@@ -0,0 +1,161 @@
+        }
+
+        const string ErrorSpyQueueName = "gw_error_spy_queue";
+
+        class Context : ScenarioContext
+        {
+            public Guid Id { get; set; }
+            public bool MessageMovedToErrorQueue { get; set; }
+            public int NumberOfRetries { get; set; }
+            public bool CustomRetryPolicyWasCalled { get; set; }
+        }
+
+        class Headquarters : EndpointConfigurationBuilder
+        {
+            public Headquarters()
+            {
+                EndpointSetup<DefaultServer>(c =>
+                {
+                    c.ScaleOut().InstanceDiscriminator("1");
+                    c.EnableFeature<Gateway>();
+                    c.EnableFeature<TimeoutManager>();
+                    c.SendFailedMessagesTo(ErrorSpyQueueName);
+                })
+                    .WithConfig<GatewayConfig>(c =>
+                    {
+                        c.Sites = new SiteCollection
+                        {
+                            new SiteConfig
+                            {
+                                Key = "SiteA",
+                                Address = "http://localhost:25999/SiteA/",
+                                ChannelType = "http"
+                            }
+                        };
+
+                        c.Channels = new ChannelCollection
+                        {
+                            new ChannelConfig
+                            {
+                                Address = "http://localhost:25999/Headquarters/",
+                                ChannelType = "http"
+                            }
+                        };
+                    });
+            }
+
+            public class AnyMessageHandler : IHandleMessages<AnyMessage>
+            {
+                public Context Context { get; set; }
+
+                public Task Handle(AnyMessage errorMessage, IMessageHandlerContext context)
+                {
+                    if (errorMessage.Id == Context.Id)
+                    {
+                        if (context.MessageHeaders.ContainsKey(Headers.Retries))
+                        {
+                            Context.NumberOfRetries = Parse(context.MessageHeaders[Headers.Retries]);
+                        }
+                        Context.MessageMovedToErrorQueue = true;
+                    }
+
+                    return Task.FromResult(0);
+                }
+            }
+        }
+
+        public class AnyMessage : IMessage
+        {
+            public Guid Id { get; set; }
+        }
+
+        class ErrorSpy : EndpointConfigurationBuilder
+        {
+            public ErrorSpy()
+            {
+                EndpointSetup<ErrorQueueSpyServer>()
+                    .CustomEndpointName(ErrorSpyQueueName);
+            }
+
+            class ErrorMessageHandler : IHandleMessages<AnyMessage>
+            {
+                public Context Context { get; set; }
```

Comment:
Use ctor injection instead (we're heading towards not supporting property injection ootb)=